src/diffusers/models/unet_2d_blocks.py

```diff
@@ -503,6 +503,29 @@ def forward(self, x):
 
 
 class UNetMidBlock2D(nn.Module):
+    """
+    A 2D UNet mid-block [`UNetMidBlock2D`] with multiple residual blocks and optional attention blocks.
+
+    Args:
+        in_channels (`int`): The number of input channels.
+        temb_channels (`int`): The number of temporal embedding channels.
+        dropout (`float`, *optional*, defaults to 0.0): The dropout rate.
+        num_layers (`int`, *optional*, defaults to 1): The number of residual blocks.
+        resnet_eps (`float`, *optional*, 1e-6 ): The epsilon value for the resnet blocks.
+        resnet_time_scale_shift (`str`, *optional*, defaults to `default`): The time scale shift for the resnet blocks.
+        resnet_act_fn (`str`, *optional*, defaults to `swish`): The activation function for the resnet blocks.
+        resnet_groups (`int`, *optional*, defaults to 32): The number of groups for the resnet blocks.
+        attn_groups (`Optional[int]`, *optional*, defaults to None): The number of groups for the attention blocks.
+        resnet_pre_norm (`bool`, *optional*, defaults to `True`): Whether to use pre-normalization for the resnet blocks.
+        add_attention (`bool`, *optional*, defaults to `True`): Whether to add attention blocks.
+        attention_head_dim (`int`, *optional*, defaults to 1): The number of attention heads.
```

Comment:
Checking with @patrickvonplaten:

Is `attention_head_dim` related to the _number_ of attention heads?

Reply:
`attention_head_dim` is the dimension of a single attention head and is used to calculate the number of attention heads in the mid block. 
https://github.com/huggingface/diffusers/blob/680e596d27e52f18c0bc9f74d20a42a2ba9865db/src/diffusers/models/unet_2d_blocks.py#L580

So while it does contribute to determining the number of attention heads, it does not actually set the number of heads. I would change this to 

```
attention_head_dim: Dimension of a single attention head. The number of attention heads is determined based on this value and the number of input channels.  
```

Reply:
Okay, now I also update the doc string for the attention_head_dim parameter.